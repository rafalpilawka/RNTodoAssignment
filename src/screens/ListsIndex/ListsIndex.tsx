import React, {FC, ReactElement} from 'react';
import {useDispatch, useSelector} from 'react-redux';
import {EmptyListComponent} from 'src/components/EmptyListComponent/EmptyListComponent';
import {List} from 'src/components/List/List';
import {Header} from 'src/components/Header/Header';
import {ScrollWrapper} from 'src/components/ScrollWrapper/ScrollWrapper';
import {removeListAction} from 'src/store/todos/todos.actions';
import {selectIndex} from 'src/store/todos/todos.selector';
import {ListsIndexT, ListType} from 'src/store/todos/todos.types';
export type ListIndexPropsT = {
  navigate(id: ListType): void;
};

export const ListsIndex: FC<ListIndexPropsT> = ({navigate}): ReactElement => {
  const index: ListsIndexT = useSelector(selectIndex);
  const dispatch = useDispatch();

  const _removeList = (id: string) => {
    dispatch(removeListAction(id));
  };

  const _renderList = () =>
    index.length ? (
      (index as Array<any>).map((el) => {
        return (
          <List
            item={el}
            removeHandler={_removeList}
            navigate={() => navigate(el)}
            key={el.id}
          />
        );
      })
    ) : (
      <EmptyListComponent variant={'LISTS'} />
    );

  return (
    <>
      <Header />
      <ScrollWrapper>{_renderList()}</ScrollWrapper>
    </>
  );
};
