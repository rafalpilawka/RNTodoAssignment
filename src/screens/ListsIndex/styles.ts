import {StyleSheet} from 'react-native';
import normalize from 'react-native-normalize';

export const styles = StyleSheet.create({
  row: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    width: '100%'
  },
});
