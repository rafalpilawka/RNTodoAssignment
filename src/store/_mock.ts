export const mockTodos = [
  {id: 'aaa', title: 'Add those todos', completed: false},
  {id: 'aaa2233', title: 'Delete second', completed: false},
  {id: 'bbnjhjhjhjh', title: 'Add third one', completed: true},
];

export const mockLists = [
  {name: 'List 1', id: '11242342144'},
  {name: 'list 2', id: '11242342122'},
  {name: 'list 3', id: '1124234211'},
];

export const mockStore = {
  index: mockLists,
  todos: mockTodos,
  loading: false,
  errors: undefined,
  idLoader: undefined,
};

export const mockInitialStore = {
  index: [],
  todos: [],
  loading: false,
  errors: undefined,
  idLoader: undefined,
};
