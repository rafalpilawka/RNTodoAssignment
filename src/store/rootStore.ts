import {all, call} from 'redux-saga/effects';
import {applyMiddleware, combineReducers, createStore, Reducer} from 'redux';
import {composeWithDevTools} from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';
import todosReducer from 'src/store/todos/todos.reducer';
import {watchTodosSaga} from 'src/store/todos/todos.saga';

const sagaMiddleware = createSagaMiddleware();

function* rootSaga() {
  yield all([
    call(watchTodosSaga),
    //OTHER WATCHERS GOES HERE
  ]);
}

// const store: Reducer<RootState> = combineReducers<RootState>({
//   todos: todosReducer,
// });

const rootStore = (state: any, action: any) => {
  return todosReducer(state, action);
};

// export interface RootState {
//   todos: ReduxState;
// }

export type RootState = ReturnType<typeof todosReducer>;

export default createStore(
  rootStore,
  composeWithDevTools(applyMiddleware(sagaMiddleware)),
);
sagaMiddleware.run(rootSaga);
