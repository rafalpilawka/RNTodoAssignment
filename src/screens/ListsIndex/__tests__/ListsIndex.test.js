import React from 'react';
import 'react-native';
import {ListsIndex} from 'src/screens/ListsIndex/ListsIndex';
import renderer from 'react-test-renderer';
import {mockStore} from 'src/store/_mock';

jest.mock('react-redux', () => ({
  useDispatch: () => () => jest.fn(),
  useSelector: (selector) => selector(mockStore),
}));
const navigate = jest.fn();

describe('ListsIndex', () => {
  describe('Render snapshot', () => {
    it('should match to snapshot', () => {
      const component = renderer
        .create(<ListsIndex navigate={navigate} />)
        .toJSON();
      expect(component).toMatchSnapshot();
    });
  });
});
