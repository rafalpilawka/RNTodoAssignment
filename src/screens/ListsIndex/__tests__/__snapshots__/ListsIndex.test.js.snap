// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListsIndex Render snapshot should match to snapshot 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "alignItems": "center",
          "backgroundColor": Object {
            "semantic": Array [
              "systemBlue",
            ],
          },
          "borderBottomWidth": 1,
          "height": 100,
          "justifyContent": "space-between",
          "marginBottom": 30,
          "paddingHorizontal": 40,
          "width": "100%",
        },
      ]
    }
  >
    <View />
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "Damascus",
          "fontSize": 40,
          "fontWeight": "bold",
        }
      }
    >
      INDEX OF TODOS
    </Text>
    <View />
  </View>,
  <RCTScrollView
    bounces={true}
    contentContainerStyle={
      Object {
        "flexGrow": 1,
        "marginBottom": 20,
        "marginHorizontal": 40,
        "paddingBottom": 40,
      }
    }
    keyboardShouldPersistTaps="always"
    showsVerticalScrollIndicator={false}
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View>
      <View
        accessible={true}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderBottomColor": "black",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "height": 160,
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          List 1
        </Text>
        <View
          accessible={true}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "red",
              "borderRadius": 100,
              "height": 60,
              "justifyContent": "center",
              "width": 60,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            x
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderBottomColor": "black",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "height": 160,
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          list 2
        </Text>
        <View
          accessible={true}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "red",
              "borderRadius": 100,
              "height": 60,
              "justifyContent": "center",
              "width": 60,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            x
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderBottomColor": "black",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "height": 160,
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          list 3
        </Text>
        <View
          accessible={true}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "red",
              "borderRadius": 100,
              "height": 60,
              "justifyContent": "center",
              "width": 60,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            x
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>,
]
`;
